_additional/frontend/step3/index.html

```diff
@@ -0,0 +1,174 @@
+        <script src="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0/js/bootstrap.min.js" integrity="sha384-JZR6Spejh4U02d8jOt6vLEHfe/JQGiRRSQQxSfFWpi1MquVdAyjUar5+76PVCmYl" crossorigin="anonymous"></script>
+    </head>
+    <body>
+        <section id="loginForm">
+            <div class="container">
+                <form id="loginForm" method="POST" action="/">
+                    <div class="form-group">
+                        <label for="nameInput">Name</label>
+                        <input type="name" class="form-control" id="nameInput" placeholder="Enter Name">
+                    </div>
+                    <button type="submit" class="btn btn-primary">Login</button>
+                </form>
+            </div>
+        </section>
+        <section class="container" id="messagePage">
+            <section id="userList" style="display:none;">
+                    <form action="#" id="logout">
+                        <button type="submit" class="btn btn-primary mb-2">Logout</button>
+                    </form> 
+                    <div class="content">
+                        <div id="users"></div>
+                    </div>
+                </section> 
+                <section id="messagesSection" style="display:none;">
+                    <div class="content">
+                        <div id="messages"></div>
+                    </div>
+                </section> 
+                <section id="textInput" class="container" style="display:none;">
+                    <form id="messageForm" method="POST">
+                        <div class="form-group">                        
+                            <input type="text" class="form-control form-control-lg" id="messageBody" placeholder="Type your message...">
+                        </div>
+                        <button type="submit" class="btn btn-primary">Send</button>                   
+                    </form>           
+                </section>
+        </section>        
+        <script>
+            var data = {};
+            var users = {};
+            var name = '';
+            var currentMessages = new Set();
+            var currentUsers = new Set();
+
+            function getUsers () {
+                $.ajax({
+                    type: 'GET',
+                    url: '/users',
+                    dataType: "JSON",
+                    data: data,
+                    xhrFields: {
+                        withCredentials: false
+                    },
+                    success: function(data) {
+                        for(var i=0; i < data.users.length; i++) {
+                            if(!currentUsers.has(data.users[i])){
+                                currentUsers.add(data.users[i]);
+                                console.log(data.users[i]);

```

Comment:
Zastanawiam się czy to do wywalennia czy do dodania w analogicznym miejscu w funkcji niżej.